apps/crn-server/test/controllers/news.test.ts

```diff
@@ -120,6 +120,164 @@ describe('News controller', () => {
 
       expect(result.items[0]!.thumbnail).toBeUndefined();
     });
+
+    test('Should query data properly when request does not have options', async () => {
+      nock(baseUrl)
+        .get(`/api/content/${appName}/news-and-events`)
+        .query({
+          q: JSON.stringify({
+            take: 8,
+            filter: { path: 'data.type.iv', op: 'ne', value: 'Tutorial' },
+            sort: [{ order: 'descending', path: 'created' }],
+          }),
+        })
+        .reply(200, newsSquidexApiResponse);
+
+      const result = await news.fetch();
+
+      expect(result).toEqual(listNewsResponse);
+    });
+
+    describe('Frequency Filter', () => {
+      test('Should query data properly when News Articles is selected', async () => {
+        nock(baseUrl)
+          .get(`/api/content/${appName}/news-and-events`)
+          .query({
+            q: JSON.stringify({
+              take: 8,
+              skip: 5,
+              filter: {
+                and: [
+                  { path: 'data.type.iv', op: 'ne', value: 'Tutorial' },
+                  {
+                    or: [
+                      {
+                        path: 'data.frequency.iv',
+                        op: 'empty',
+                        value: null,
+                      },
+                      {
+                        path: 'data.frequency.iv',
+                        op: 'in',
+                        value: ['CRN Quarterly', 'News Articles'],
+                      },
+                    ],
+                  },
+                ],
+              },
+              sort: [{ order: 'descending', path: 'created' }],
+            }),
+          })
+          .reply(200, newsSquidexApiResponse);
```

Comment:
Although this `newsSquidexApiResponse` does not have appropriate data for this test, my understanding was that these tests are checking the query not the functionality itself. That's why I didn't change the `newsSquidexApiResponse` throughout the tests. But I am not sure if that's the best thing to do. What do you all think? :)

Reply:
thats fine, news is a very old one which still has those 'test-it-all' kind of tests so we're going to address it later when we split the controller into two layers